velox/type/Variant.cpp

```diff
@@ -183,6 +183,15 @@ std::string variant::toJson(const TypePtr& type) const {
     return "null";
   }
 
+  if (type != nullptr) {
```

Comment:
This check should be `VELOX_USER_CHECK(type != nullptr)` and at the top.

Reply:
I see why you need this check. `type` can be a `nullptr`.
It is not ideal to allow type to be a nullptr. It makes the implementation clunky and bug prone.
Example: `type->isDate()` check below is missing the nullptr check.
Let me look into this further.